packages/contractkit/src/providers/celo-private-keys-subprovider.ts

```diff
@@ -112,9 +123,15 @@ export class CeloPrivateKeysWalletProvider extends PrivateKeyWalletSubprovider {
       }
     }
 
-    if (txParams.gasPrice == null) {
+    if (isEmpty(txParams.gasPrice)) {
       txParams.gasPrice = await this.getGasPrice()
     }
+    debug('Gas price for the transaction is %s', txParams.gasPrice)
+
+    if (isEmpty(txParams.gas)) {
+      txParams.gas = String(Math.round((await this.getGasEstimate(txParams)) * gasInflationFactor))
```

Comment:
this would only work if the gasCurrency is cGold.

Reply:
Fixed. I could not test it  though since fetching the address of the stable token fails during testing.